### PR TITLE
Refactorings

### PR DESCRIPTION
**Pull request by genmad**
some refactorings I thought were useful

### CODE REVIEW

File: src/main/java/hudson/plugins/filesystem_scm/ChangelogSet.java [hunk 1]
```diff
@@ -115,14 +115,14 @@ public ChangelogSet parse(Run<?,?> build, java.io.File file) throws FileNotFound
...
+		public void save(ChangelogSet changeLogSet, File changelogFile) throws FileNotFoundException {
 			FileOutputStream out = null;
 			try {
-				out = new FileOutputStream(file);
+				out = new FileOutputStream(changelogFile);
 				xstream.toXML(changeLogSet, out);
 			} finally {
 				IOUtils.closeQuietly(out);
-			}
+			}
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/filesystem_scm/ChangelogSet.java]: not sure why would you need these tabs

File: src/main/java/hudson/plugins/filesystem_scm/FSSCM.java [hunk 2]
```diff
@@ -84,11 +84,12 @@ public FSSCM(String path, boolean clearWorkspace, boolean copyHidden, FilterSett
...
     @Deprecated
     public FSSCM(String path, boolean clearWorkspace, boolean copyHidden, boolean filterEnabled, boolean includeFilter, String[] filters) {
-        this.path = path;
-        this.clearWorkspace = clearWorkspace;
-        this.copyHidden = copyHidden;
+    		this(path, clearWorkspace, copyHidden, createFilterSettings(filterEnabled, includeFilter, filters));
+    }
 
-        if (filterEnabled) {
+	private static FilterSettings createFilterSettings(boolean filterEnabled, boolean includeFilter, String[] filters) {
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/filesystem_scm/FSSCM.java]: tabs vs. spaces

**Comment by oleg-nenashev** [comment]
@genmad It does a full refactoring of the codebase, but I am fine with that 👍 . Just need the merge conflict to be resolved

**Comment by genmad** [comment]
@oleg-nenashev I'm at solving the merge conflict, but still struggling with the tools. As I said my first time working under GitHub. May still take a while.

**Comment by oleg-nenashev** [comment]
@genmad I can release the fix without waiting for this PR if you prefer

**Comment by genmad** [comment]
I don't care. I don't have the problem of the fix anymore. It is up to you.

**Comment by genmad** [comment]
@oleg-nenashev this branch seems ready to go, merge conflict solved.

**Pull request merged**